lab_5_scrapper/scrapper.py

```diff
@@ -2,73 +2,172 @@
+# from core_utils.constants import (CRAWLER_CONFIG_PATH)
+
+
+class IncorrectSeedURLError(Exception):
+    """
+    Raises when seed URL does not match standard pattern "https?://w?w?w?.
+    """
+    pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    """
+    Raises when total number of articles is out of range from 1 to 150
+    """
+    pass
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    """
+    Raises when total number of articles to parse is not integer
+    """
+    pass
+
+
+class IncorrectHeadersError(Exception):
+    """
+    Raises when total number of articles to parse is not integer
+    """
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    """
+    Raises when encoding does not specified as a string
+    """
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    """
+    Raises when timeout value is not a positive integer less than 60
+    """
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    """
+    Raises when verify certificate value does not either be True or False
+    """
+    pass
 
 
 class Config:
     """
     Unpacks and validates configurations
     """
-
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._config_data = self._extract_config_content()
+        self._validate_config_content()
+        self._seed_urls = self._config_data.seed_urls
+        self._num_articles = self._config_data.total_articles_to_find_and_parse
+        self._headers = self._config_data.headers
+        self._encoding = self._config_data.encoding
+        self._timeout = self._config_data.timeout
+        self._verify_certificate = self._config_data.should_verify_certificate
+        self._headless_mode = self._config_data.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+        return ConfigDTO(**content)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+
+        seed_urls = content['seed_urls']
+        headers = content['headers']
+        total_articles_to_find_and_parse = content['total_articles_to_find_and_parse']
+        encoding = content['encoding']
+        timeout = content['timeout']
+        should_verify_certificate = content['should_verify_certificate']
+
+        if not isinstance(seed_urls, list)\
+                or not not all(isinstance(url, str) for url in seed_urls)\
+                or not all(re.match('https?://w?w?w?.', url) for url in seed_urls):
+            raise IncorrectSeedURLError
```

Comment:
made a check which measures length of this list (len(self._seed_urls) == 0)

Reply:
you do not need to write `==0`, `if self._seed_urls` is enough (try to think why)

Reply:
yes, I saw this kind of check but chose this one for some reason)
if I'm right, "if self._seed_urls" gives us bool result and in case seed_urls is an empty list it will give us False result which will cause the error to rise